code/controllers/subsystem/achievements.dm

```diff
@@ -47,6 +48,23 @@ SUBSYSTEM_DEF(achievements)
 	qdel(ridOldChieves)
 	return ..()
 
+/datum/controller/subsystem/achievements/fire(resumed)
+	//The solar panel achievement
+	if(!CE)
+		for(var/x in GLOB.player_list)
+			if(ishuman(x))
+				var/mob/living/carbon/human/H = x
+				if(H?.mind.assigned_role == "Chief Engineer")
+					CE = H
+					break
+	else
+		for(var/n in SSmachines.powernets)
+			var/datum/powernet/net = n
+			if(net.z == 2) // If the powernet is on the station z-level
+				if(net.avail >= 3000 && CE.stat != DEAD && CE.client) // If there's 3 MW available (Value is in kW)
+					unlock_achievement(/datum/achievement/engineering/scotty, CE.client)
```

Comment:
This whole thing should honestly just be [here](https://github.com/yogstation13/Yogstation/blob/1bee27921e26d401c34671568ae22c7948ed7dc8/code/modules/power/powernet.dm#L95) instead

Reply:
Yeah that's exactly what I was alluding to when I said I didn't want to waste ``SSmachine``'s time; /datum/powernet/proc/reset() gets called only by SSmachine. I'd rather have this whole thing not be done on its time, when SSachievements already has its very own subsystem to work with.